tests/src/Functional/RequestTest.php

```diff
@@ -0,0 +1,982 @@
+        'type' => 'text',
+      ])
+        ->setCardinality(1)
+        ->save();
+      FieldConfig::create([
+        'entity_type' => $entity_type,
+        'field_name' => 'field_test_' . $entity_type,
+        'bundle' => 'camelids',
+      ])
+        ->setLabel('Test field')
+        ->setTranslatable(FALSE)
+        ->save();
+    }
+    // @todo Schemata Routes aren't rebuilt after new content type.
+    $this->container->get('module_installer')->install(['schemata', 'schemata_json_schema'], TRUE);
+    $this->drupalLogin($this->drupalCreateUser(['access schemata data models']));
+  }
+
+  /**
+   * Tests schemata requests.
+   */
+  public function testRequests() {
+    /** @var \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager */
+    $entity_type_manager = $this->container->get('entity_type.manager');
+    $options = [
+      'query' => [
+        '_format' => 'schema_json',
+        '_describes' => 'json',
+      ],
+    ];
+    foreach ($entity_type_manager->getDefinitions() as $entity_type_id => $entity_type) {
+
+      $response = $this->request('GET', Url::fromRoute("schemata.$entity_type_id", [], $options), []);
+      $this->checkExpectResponse($response, $entity_type_id);
+      if ($bundle_type = $entity_type->getBundleEntityType()) {
+        $bundles = $entity_type_manager->getStorage($bundle_type)->loadMultiple();
+        foreach ($bundles as $bundle) {
+          $response = $this->request('GET', Url::fromRoute("schemata.$entity_type_id:{$bundle->id()}", [], $options), []);
+          $this->checkExpectResponse($response, $entity_type_id, $bundle->id());
+        }
+      }
+    }
+  }
+
+  /**
+   * Performs a HTTP request. Wraps the Guzzle HTTP client.
+   *
+   * Why wrap the Guzzle HTTP client? Because we want to keep the actual test
+   * code as simple as possible, and hence not require them to specify the
+   * 'http_errors = FALSE' request option, nor do we want them to have to
+   * convert Drupal Url objects to strings.
+   *
+   * We also don't want to follow redirects automatically, to ensure these tests
+   * are able to detect when redirects are added or removed.
+   *
+   * @see \GuzzleHttp\ClientInterface::request()
+   *
+   * @param string $method
+   *   HTTP method.
+   * @param \Drupal\Core\Url $url
+   *   URL to request.
+   * @param array $request_options
+   *   Request options to apply.
+   *
+   * @return \Psr\Http\Message\ResponseInterface
+   */
+  protected function request($method, Url $url, array $request_options) {
+    $request_options[RequestOptions::HTTP_ERRORS] = FALSE;
+    $request_options[RequestOptions::ALLOW_REDIRECTS] = FALSE;
+    $request_options = $this->decorateWithXdebugCookie($request_options);
+    $client = $this->getSession()->getDriver()->getClient()->getClient();
+    return $client->request($method, $url->setAbsolute(TRUE)->toString(), $request_options);
+  }
+
+  /**
+   * Adds the Xdebug cookie to the request options.
```

Comment:
AFAICT it forwards all cookies from the request launching the tests, not just Xdebug's.